compiler-core/src/metadata.rs

```diff
@@ -1,10 +1,98 @@
 //! Seriaisation and deserialisation of Gleam compiler metadata into binary files
-//! using the Cap'n Proto schema.
-
-mod module_decoder;
-mod module_encoder;
+//! using the Serde and Bincode
 
 #[cfg(test)]
 mod tests;
 
-pub use self::{module_decoder::ModuleDecoder, module_encoder::ModuleEncoder};
+use crate::type_::{Type, TypeVar};
+use crate::uid::UniqueIdGenerator;
+use crate::{
+    type_::{Module, TypeConstructor},
+    Result,
+};
+use std::cell::RefCell;
+use std::sync::Arc;
+
+pub fn encode(data: &Module) -> Result<Vec<u8>> {
+    let span = tracing::info_span!("metadata");
+    let _enter = span.enter();
+
+    let config = bincode::config::standard();
+    let buffer =
+        bincode::serde::encode_to_vec(data, config).expect("failed to encode metadata (bincode)");
+
+    Ok(buffer)
+}
+
+fn undo_links(id_generator: &UniqueIdGenerator, type_var: TypeVar) -> Type {
+    match type_var {
+        TypeVar::Link { type_ } => decode_type(id_generator, &type_),
+        TypeVar::Generic { .. } => Type::Var {
+            type_: Arc::new(RefCell::new(TypeVar::Generic {
+                id: id_generator.next(),
+            })),
+        },
+        TypeVar::Unbound { .. } => Type::Var {
```

Comment:
This should never happen so let's panic if there is an unbound var in the cache

Reply:
panicing now using `panic!()`